src/Raven.Client/Documents/Subscriptions/AbstractSubscriptionWorker.cs

```diff
@@ -404,6 +404,8 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
                     DatabaseDoesNotExistException.ThrowWithMessage(_dbName, connectionStatus.Message);
                 else if (connectionStatus.Exception.Contains(nameof(NotSupportedInShardingException)))
                     throw new NotSupportedInShardingException(connectionStatus.Message);
+                else if (connectionStatus.Exception.Contains(nameof(DatabaseDisabledException)))
```

Comment:
sth that we need in 5.4 maybe?

Reply:
It seems to not happen on non-sharded because when the database is disabled the server throws the error in the `HandlePath` func, before the subscription connection is established. `AssertConnectionState` handles exceptions reported in `ReportExceptionToClient` which only happens after a subscription connection has been established.
When running the test that disables the db *during* the subscription, the server reports a connection error rather than disabled error.
So it's possible that there is some path or scenario where the server will `ReportExceptionToClient` with a disabled error, but it doesn't seem to happen in these scenarios..
I could add it in 5.4 anyway if we want to cover any unknown cases